rocAL/rocAL/source/bounding_box_graph.cpp

```diff
@@ -20,12 +20,101 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    {
+        float _dst_to_src_width_ratio = roi_width[i] / float(original_width[i]);
+        float _dst_to_src_height_ratio = roi_height[i] / float(original_height[i]);
+        unsigned bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        std::vector<int> labels_buf(bb_count);
+        std::vector<float> coords_buf(bb_count * 4);
+        memcpy(labels_buf.data(), input_meta_data->get_bb_labels_batch()[i].data(), sizeof(int) * bb_count);
+        memcpy(coords_buf.data(), input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        BoundingBoxCords bb_coords;
+        BoundingBoxCord temp_box;
+        temp_box.l = temp_box.t = temp_box.r = temp_box.b = 0;
+        BoundingBoxLabels bb_labels;
+        float mask_data[MAX_BUFFER];
+        int poly_count[bb_count];
+        std::vector<int> poly_size;
+        MaskCords mask_coords;
+        coords mask_cord;
+        std::vector<float> mask;
+        int idx = 0, index = 1;
+        if (segmentation)
+        {
+            auto ptr = mask_data;
+            for (unsigned int object_index = 0; object_index < bb_count; object_index++)
+            {
+                unsigned polygon_count = input_meta_data->get_mask_cords_batch()[i][object_index].size();
+                poly_count[object_index] = polygon_count;
+                for (unsigned int polygon_index = 0; polygon_index < polygon_count; polygon_index++)
+                {
+                    unsigned polygon_size = input_meta_data->get_mask_cords_batch()[i][object_index][polygon_index].size();
+                    poly_size.push_back(polygon_size);
+                    memcpy(ptr, input_meta_data->get_mask_cords_batch()[i][object_index][polygon_index].data(), sizeof(float) * input_meta_data->get_mask_cords_batch()[i][object_index][polygon_index].size());
+                    ptr += polygon_size;
+                }
+            }
+            for (unsigned int loop_index_1 = 0, k = 0; loop_index_1 < poly_size.size(); loop_index_1++)
+            {
+                for (int loop_idx_2 = 0; loop_idx_2 < poly_size[loop_index_1]; loop_idx_2 += 2, idx += 2)
+                {
+                    mask.push_back(mask_data[idx] * _dst_to_src_width_ratio);
+                    mask.push_back(mask_data[idx + 1] * _dst_to_src_height_ratio);
+                }
+                mask_cord.push_back(mask);
+                mask.clear();
+                if (poly_count[k] == index++)
+                {
+                    mask_coords.push_back(mask_cord);
+                    mask_cord.clear();
+                    k++;
+                    index = 1;
+                }
+            }
+        }
+
+        int m = 0;
+        for (uint j = 0; j < bb_count; j++)
+        {
+            BoundingBoxCord box;
+            box.l = coords_buf[m++];
+            box.t = coords_buf[m++];
+            box.r = coords_buf[m++];
```

Comment:
Declare coords_buf as BoundingBoxCords type & directly push to bb_coords instead of assigning to box , then pushing it

Refer meta_node_flip.cpp for reference